pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -507,14 +507,22 @@ func (c *Controller) updateShootStatusDeleteSuccess(o *operation.Operation) erro
 	}
 	o.Shoot.Info = newShoot
 
-	// Remove finalizer
-	finalizers := sets.NewString(o.Shoot.Info.Finalizers...)
-	finalizers.Delete(gardencorev1alpha1.GardenerName)
-	o.Shoot.Info.Finalizers = finalizers.List()
-	newShoot, err = c.k8sGardenClient.GardenCore().CoreV1alpha1().Shoots(o.Shoot.Info.Namespace).Update(o.Shoot.Info)
+	// Remove finalizer with retry on conflict
+	newShoot, err = kutil.TryUpdateShoot(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, o.Shoot.Info.ObjectMeta, func(curShoot *gardencorev1alpha1.Shoot) (*gardencorev1alpha1.Shoot, error) {
```

Comment:
Can we re-use https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/utils/finalizers.go#L51?